src/mono/wasm/runtime/binding_support.js

```diff
@@ -1744,51 +1773,54 @@ var BindingSupportLib = {
 		}
 	},
     mono_wasm_set_object_property: function (js_handle, property_name, value, createIfNotExist, hasOwnProperty, is_exception) {
+		var valueRoot = MONO.mono_wasm_new_root (value);
+		try {
+			BINDING.bindings_lazy_init ();
+			var requireObject = BINDING.mono_wasm_require_handle (js_handle);
+			if (!requireObject) {
+				setValue (is_exception, 1, "i32");
+				return BINDING.js_string_to_mono_string ("Invalid JS object handle '" + js_handle + "'");
+			}
 
-		BINDING.bindings_lazy_init ();
-
-		var requireObject = BINDING.mono_wasm_require_handle (js_handle);
-		if (!requireObject) {
-			setValue (is_exception, 1, "i32");
-			return BINDING.js_string_to_mono_string ("Invalid JS object handle '" + js_handle + "'");
-		}
-
-		var property = BINDING.conv_string (property_name);
-		if (!property) {
-			setValue (is_exception, 1, "i32");
-			return BINDING.js_string_to_mono_string ("Invalid property name object '" + property_name + "'");
-		}
+			var property = BINDING.conv_string (property_name);
```

Comment:
Should that be rooted too ? Let's make it consistent with `method_name` in `mono_wasm_invoke_js_with_args`. And all other usages of `conv_string`. That's the one which does interning ?

Reply:
Good point, while it should normally be an interned string (and thus not require a root), it might not be. So I need to update all of these to root the names.